calico_containers/pycalico/ipam.py

```diff
@@ -76,7 +76,7 @@ def _compare_and_swap_block(self, block):
             _log.debug("CAS Update block %s", block)
             try:
                 self.etcd_client.update(block.update_result())
-            except ValueError:
+            except EtcdValueError:
```

Comment:
To be as specific as possible, shouldn't these all be catching `EtcdCompareFailed`?